tests/constraint/test_NonColliding.cpp

```diff
@@ -35,17 +35,11 @@ class NonCollidingTest : public ::testing::Test
     auto boxNode = mBox->createJointAndBodyNodePair<FreeJoint>().second;
     Eigen::Vector3d boxSize(0.5, 0.5, 0.5);
     std::shared_ptr<BoxShape> boxShape(new BoxShape(boxSize));
-    auto shapeNode = boxNode->createShapeNodeWith
-                      <VisualAspect, CollisionAspect, DynamicsAspect>(boxShape);
```

Comment:
You shouldn't remove the whole line, but just `shapeNode` variable. `createShapeNodeWith<...>()` creates `ShapeNode` in `boxNode` and returns the pointer. That means this function changes the internal state of `bodyNode`, which is necessary for this test.